x-pack/legacy/plugins/apm/public/components/shared/LocalFilters/Filter.tsx

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useCallback, useMemo, useEffect } from 'react';
+import {
+  EuiTitle,
+  EuiButtonEmpty,
+  EuiPopover,
+  EuiSelectable,
+  EuiSpacer,
+  EuiHorizontalRule,
+  EuiText,
+  EuiButton,
+  EuiBadge,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiIcon,
+  EuiFlexGroup
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import styled from 'styled-components';
+
+const Popover = styled(EuiPopover).attrs({
+  anchorClassName: 'anchor'
+})`
+  .anchor {
+    display: block;
+  }
+`;
+
+const SelectContainer = styled.div`
+   {
+    width: 300px;
+  }
+`;
+
+const Counter = styled.div`
+  border-radius: ${theme.euiBorderRadius};
+  background: ${theme.euiColorLightShade};
+  padding: 0 ${theme.paddingSizes.xs};
+`;
+
+const BadgeText = styled.div`
+   {
+    display: inline-block;
+    max-width: 140px;
```

Comment:
Use units instead of literal pixel values. This makes it easier to retain the horizontal rhythm and avoid "of by 1px" issues.

Reply:
Aye, kind of just winged this 😅